pkg/virt-controller/services/template_test.go

```diff
@@ -428,6 +428,56 @@ var _ = Describe("Template", func() {
 				Expect(pod.Spec.ImagePullSecrets[1].Name).To(Equal("pull-secret-1"))
 			})
 		})
+
+		Context("with slirp interface", func() {
+			It("Should have empty port list in the pod manifest", func() {
+				slirpInterface := v1.InterfaceSlirp{}
+				domain := v1.DomainSpec{}
+				domain.Devices.Interfaces = []v1.Interface{{Name: "testnet", InterfaceBindingMethod: v1.InterfaceBindingMethod{Slirp: &slirpInterface}}}
+				vmi := v1.VirtualMachineInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testvmi", Namespace: "default", UID: "1234",
+					},
+					Spec: v1.VirtualMachineInstanceSpec{Domain: domain},
+				}
+
+				pod, err := svc.RenderLaunchManifest(&vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(len(pod.Spec.Containers)).To(Equal(1))
+				Expect(len(pod.Spec.Containers[0].Ports)).To(Equal(0))
+			})
+			It("Should create a port list in the pod manifest", func() {
+				slirpInterface := v1.InterfaceSlirp{}
+				ports := []v1.Port{{Name: "http", Port: 80}, {Protocol: "UDP", Port: 80}, {Port: 90}, {Name: "other-http", Port: 80}}
+				domain := v1.DomainSpec{}
+				domain.Devices.Interfaces = []v1.Interface{{Name: "testnet", Ports: ports, InterfaceBindingMethod: v1.InterfaceBindingMethod{Slirp: &slirpInterface}}}
```

Comment:
Please add a case, or modify this one, to execute with multiple interfaces, to check all ports for all interfaces are combined and passed forward. Also, since you allow duplicate ports for the same interface in API, maybe also add a case to pass duplicate ports.